src/ui/simple_ui/ask_hash.rs

```diff
@@ -55,6 +98,35 @@ pub fn ask_hash(args: &BurnArgs, cf: CompressionFormat) -> anyhow::Result<Option
     Ok(Some(hash_result))
 }
 
+fn find_hash(input: &PathBuf) -> Option<(HashAlg, &str, Vec<u8>)> {
+    for (alg, hash_file) in HASH_FILES {
+        let hash_filepath = input.parent()?.join(hash_file);
+        if let Ok(true) = fs::exists(&hash_filepath) {
```

Comment:
instead of checking for `fs::exists`, why not just open the file and skip if it's unable to be read?